weave/wandb_interface/wandb_lite_run.py

```diff
@@ -156,16 +189,37 @@ def pusher(self) -> FilePusher:
 
         return self._pusher
 
+    def log_artifact(self, artifact: "Artifact") -> typing.Optional[dict]:
+        saver = artifact_saver.ArtifactSaver(
+            api=self.i_api,
+            digest=artifact.digest,
+            manifest_json=artifact.manifest.to_manifest_json(),
+            file_pusher=self.pusher,
+            is_user_created=False,
+        )
+        return saver.save(
+            type=artifact.type,
+            name=artifact.name,
+            client_id=artifact._client_id,
+            sequence_client_id=artifact._sequence_client_id,
+            metadata=artifact.metadata,
+            description=artifact.description or None,
+            aliases=artifact._aliases,
+            use_after_commit=False,
+            distributed_id=None,
+            finalize=True,
+            incremental=False,
+            history_step=0,
+            base_id=artifact._base_id or None,
+        )
+
     def log(self, row_dict: dict) -> None:
         stream = self.stream
         row_dict = {
             **row_dict,
             "_timestamp": datetime.datetime.utcnow().timestamp(),
         }
-        if not self._use_async_file_stream:
-            row_dict["_step"] = self._step
-        self._step += 1
-        stream.push("wandb-history.jsonl", json.dumps(row_dict))
+        stream.push("wandb-history.jsonl", util.json_dumps_safer_history(row_dict))
```

Comment:
This serializes everything the wandb/sdk way.

Reply:
how does this handle run files? If a media type is in the row_dict, they will often need to "bind_to_run" and save their files to the run